### PR TITLE
xtensa-build-zephyr.py: make --deployable-build the default

### PR DESCRIPTION
**Pull request by marc-hb**
_[OLD commit message]_

Everyone should use deployable builds by default.

------

_[NEW commit message]_

Every Linux developer should use deployable builds by default.

Until Peter Ujfalusi's very recent work in this script, we had a
complete `/lib/firmware/` structure disconnect between the IPC4 output
of this script and the IPC4 expectations of the Linux kernel. To
workaround this disconnect, every CI and Linux developer used to
implement duplicate and inconsistent firmware deployment hacks.

People crafting sof-bin releases also had to organize IPC4 releases
manually, which was extremely error-prone and with limited test
coverage (Thanks Kai and Mengdong!)

Now that Peter gracefully fixed the layout, documented it in sof-docs
and implemented it in this script, the time for all Linux developers to
drop their inconsistent deployment hacks is overdue. All these hacks
must be replaced with a simple, one-line recursive copy which makes sure
the layout committed in version control is constantly tested by
everyone.

So, make deployable builds the new default.

The new default will also help with sof-bin releases, making sure they
use a well tested /lib/firmware/ layout.

The --no-deployable-build was recently introduced to help minimize
disruption and migration effort for people and automation who do NOT use
Linux. The `/lib/firmware/` directory structure is irrelevant outside
Linux (but everyone is of course free to choose it)

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
@lrudyX , @wszypelt , all tests failed like this:
```
E FileNotFoundError: [Errno 2] No such file or directory:
'C:\\Testing\\Builds\\pull-8781-merge_1931053\\
   cSOF_zephyr\\LNL\\build-sof-staging\\sof\\community\\sof-lnl.ri'
```

Could you please add the new `--no-deployable-build` option where appropriate? This would make QB ignore this PR and stick to the older behavior. It can be done before this PR is merged, `--no-deployable-build` is already available now. This PR only flips the default value.

**Comment by wszypelt** [comment]
Hi @marc-hb first thing is the link you provide shows me a blank page
(https://sof-ci.01.org/sof-pr-viewer/#/build/PR8781/build13454079)
as for the "--no-deployable-build option"  i need to discuss it with the team

**Comment by abonislawski** [comment]
If everyone should use deployable build then why QB shouldn't?

**Comment by marc-hb** [comment]
> If everyone should use deployable build then why QB shouldn't?

I'm fine with that too but I expect this to take much more work and take much longer. The "deployable" layout matters only when you run a Linux kernel and these tests do not.

**Comment by marc-hb** [comment]
> Hi @marc-hb first thing is the link you provide shows me a blank page
(https://sof-ci.01.org/sof-pr-viewer/#/build/PR8781/build13454079)

You need to extract the number 13454079 and re-use it on the internal site. We lost the public export of these results a long time ago unfortunately.

**Comment by marc-hb** [comment]
> @marc-hb can you say why everyone should use and any possible exceptions.

I just increased the commit message length from 1 line to 28 lines. Hope this clarifies!

**Comment by wszypelt** [comment]
@marc-hb --no-deployable-build option added where appropriate. PR passed all tests

**Comment by marc-hb** [comment]
Just one suspend/resume failure in https://sof-ci.01.org/sofpr/PR8781/build2242/devicetest/index.html and one pause resume "MAX" failure in https://sof-ci.01.org/sofpr/PR8781/build2241/devicetest/index.html

zephyr main branch incompatibility is known.

Nothing related to this PR, good to merge.

**Comment by marc-hb** [comment]
After second thought, back to draft until the current regressions and zephyr updates are resolved. Enough moving pieces already.

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
Postponing again until after the latest, HWMv2 hurdle #8913

**Comment by marc-hb** [comment]
Frequent panic #8908 still observed in https://sof-ci.01.org/sofpr/PR8781/build3249/devicetest/index.html.

Postponing one hopefully last time.

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
Only one known and totally unrelated failure at https://sof-ci.01.org/sofpr/PR8781/build3478/devicetest/index.html?model=MTLP_RVP_NOCODEC&testcase=multiple-pause-resume-50 , all the rest of that test plan is green.

Only one, totally unrelated suspend/resume failure in https://sof-ci.01.org/sofpr/PR8781/build3479/devicetest/index.html

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
@ujfalusi @lyakh Can you check? We'd need on more approval, this is ready to go.

**Review by ujfalusi — APPROVED** [review]

**Pull request merged**
